bin/node/src/config.rs

```diff
@@ -1,35 +1,71 @@
+use std::path::Path;
+
 use miden_node_block_producer::config::BlockProducerConfig;
 use miden_node_rpc::config::RpcConfig;
 use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::Endpoint;
 use serde::{Deserialize, Serialize};
 
 /// Node top-level configuration.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+#[derive(Clone, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
 pub struct NodeConfig {
-    pub block_producer: Option<BlockProducerConfig>,
-    pub rpc: Option<RpcConfig>,
-    pub store: Option<StoreConfig>,
+    pub block_producer: BlockProducerConfig,
+    pub rpc: RpcConfig,
+    pub store: StoreConfig,
 }
 
-impl Default for NodeConfig {
-    fn default() -> Self {
-        Self {
-            block_producer: Some(Default::default()),
-            rpc: Some(Default::default()),
-            store: Some(Default::default()),
+impl NodeConfig {
+    pub fn load(config_file: impl AsRef<Path>) -> figment::Result<Self> {
```

Comment:
The `figment::Result` is required to properly test using `figment::jail` which is a pity as this adds `figment` to our dependencies.

This is probably fine though - I imagination a binary should contain configuration code :). Alternatively we create some better abstractions or hide `figment` somehow, but I don't really think that's worth the effort. lmk.

Reply:
Yep - I think that's fine. I left a comment inline about potentially using `Figment` directly from the node rather than having a utility function for it in `miden-node-utils`.

Reply:
I've reworked this so that this is no longer necessary.